tensorflow_addons/seq2seq/tests/attention_wrapper_test.py

```diff
@@ -44,217 +43,241 @@ def setUp(self):
         self.query = np.random.randn(self.batch, self.units).astype(np.float32)
         self.state = np.random.randn(self.batch, self.timestep).astype(np.float32)
 
-    @parameterized.named_parameters(
-        ("luong", wrapper.LuongAttention),
-        ("luong_monotonic", wrapper.LuongMonotonicAttention),
-        ("bahdanau", wrapper.BahdanauAttention),
-        ("bahdanau_monotonic", wrapper.BahdanauMonotonicAttention),
-    )
-    def test_attention_shape_inference(self, attention_cls):
-        attention = attention_cls(self.units, self.memory)
-        attention_score = attention([self.query, self.state])
-        self.assertLen(attention_score, 2)
-        self.assertEqual(attention_score[0].shape, (self.batch, self.timestep))
-        self.assertEqual(attention_score[1].shape, (self.batch, self.timestep))
-
-    @parameterized.named_parameters(
-        ("luong", wrapper.LuongAttention),
-        ("luong_monotonic", wrapper.LuongMonotonicAttention),
-        ("bahdanau", wrapper.BahdanauAttention),
-        ("bahdanau_monotonic", wrapper.BahdanauMonotonicAttention),
-    )
-    def test_get_config(self, attention_cls):
-        attention = attention_cls(self.units, self.memory)
-        config = attention.get_config()
-
-        attention_from_config = attention_cls.from_config(config)
-        config_from_clone = attention_from_config.get_config()
 
-        self.assertDictEqual(config, config_from_clone)
-
-    @parameterized.named_parameters(
-        ("luong", wrapper.LuongAttention),
-        ("luong_monotonic", wrapper.LuongMonotonicAttention),
-        ("bahdanau", wrapper.BahdanauAttention),
-        ("bahdanau_monotonic", wrapper.BahdanauMonotonicAttention),
-    )
-    def test_layer_output(self, attention_cls):
-        attention = attention_cls(self.units, self.memory)
-        score = attention([self.query, self.state])
-        self.evaluate(tf.compat.v1.variables_initializer(attention.variables))
-
-        score_val = self.evaluate(score)
-        self.assertLen(score_val, 2)
-        self.assertEqual(score_val[0].shape, (self.batch, self.timestep))
-        self.assertEqual(score_val[1].shape, (self.batch, self.timestep))
-
-    @parameterized.named_parameters(
-        ("luong", wrapper.LuongAttention),
-        ("luong_monotonic", wrapper.LuongMonotonicAttention),
-        ("bahdanau", wrapper.BahdanauAttention),
-        ("bahdanau_monotonic", wrapper.BahdanauMonotonicAttention),
+@pytest.mark.parametrize(
```

Comment:
Can we extra this annotation and reuse it? Having so many duplication seems to be a issue for future maintainence.